contracts/TokenConverter/RiskFundConverter.sol

```diff
@@ -213,13 +210,26 @@ contract RiskFundConverter is AbstractTokenConverter {
     /// @param tokenAddress Address of the token
     function getPools(address tokenAddress) internal view returns (address[] memory) {
         address[] memory pools = IPoolRegistry(poolRegistry).getPoolsSupportedByAsset(tokenAddress);
-        uint256 poolsLength = pools.length;
-        address[] memory poolsWithCore = new address[](poolsLength + 1);
 
-        for (uint256 i; i < poolsLength; ++i) {
-            poolsWithCore[i] = pools[i];
+        if (IComptroller(corePoolComptroller).markets(tokenAddress)) {
```

Comment:
I think this doesn't seem right because the keys in the `markets` variable are the vTokens, not the underlying tokens. We don't have the needed data structure in the Core Comptroller, so I think we have to iterate on the `allMarkets` array checking the underlying tokens, in the short term, and the in future update the Core Comptroller to add the right mapping underlying -> bool